src/serac/physics/utilities/finite_element_state.hpp

```diff
@@ -45,78 +46,49 @@ inline bool is_vector_valued(const GeneralCoefficient& coef)
-    mfem::Ordering::Type ordering = mfem::Ordering::byVDIM;
-    /**
-     * @brief The name of the field encapsulated by the state object
-     */
-    std::string name = "";
-    /**
-     * @brief Whether the GridFunction should be allocated (and owned by the FEState object)
-     */
-    bool alloc_gf = true;
-  };
-
-  /**
-   * Main constructor for building a new state object
+   * Main constructor for building a new finite element state
    * @param[in] mesh The problem mesh (object does not take ownership)
    * @param[in] options The options specified, namely those relating to the order of the problem,
    * the dimension of the FESpace, the type of FEColl, the DOF ordering that should be used,
    * and the name of the field
    */
-  FiniteElementState(
-      mfem::ParMesh& mesh,
-      Options&&      options = {
-          .order = 1, .vector_dim = 1, .coll = {}, .ordering = mfem::Ordering::byVDIM, .name = "", .alloc_gf = true});
+  FiniteElementState(mfem::ParMesh& mesh, Options&& options = {})
+      : FiniteElementVector(mesh, std::forward<FiniteElementVector::Options>(options)){};
 
   /**
    * @brief Minimal constructor for a FiniteElementState given an already-existing field
    * @param[in] mesh The problem mesh (object does not take ownership)
    * @param[in] gf The field for the state to create (object does not take ownership)
    * @param[in] name The name of the field
    */
-  FiniteElementState(mfem::ParMesh& mesh, mfem::ParGridFunction& gf, const std::string& name = "");
+  FiniteElementState(mfem::ParMesh& mesh, mfem::ParGridFunction& gf, const std::string& name = "")
+      : FiniteElementVector(mesh, gf, name){};
 
   /**
-   * @brief Minimal constructor for a FiniteElementState given an already-existing state
+   * @brief Minimal constructor for a FiniteElementState given a finite element space
    * @param[in] mesh The problem mesh (object does not take ownership)
-   * @param[in] fe_state The state for the new state to copy
+   * @param[in] space The space to use for the finite element state. This space is deep copied into the new FE state
    * @param[in] name The name of the field
    */
-  FiniteElementState(mfem::ParMesh& mesh, FiniteElementState& fe_state, const std::string& name = "");
+  FiniteElementState(mfem::ParMesh& mesh, mfem::ParFiniteElementSpace& space, const std::string& name = "")
+      : FiniteElementVector(mesh, space, name){};
 
   /**
-   * Returns the MPI communicator for the state
+   * @brief Minimal constructor for a FiniteElementState given an already-existing state
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] fe_state The state for the new state to copy
+   * @param[in] name The name of the field
    */
-  MPI_Comm comm() const { return detail::retrieve(space_).GetComm(); }
+  FiniteElementState(mfem::ParMesh& mesh, FiniteElementState& fe_state, const std::string& name = "")
+      : FiniteElementVector(mesh, fe_state.gridFunc(), name){};
```

Comment:
[this](https://github.com/LLNL/serac/pull/554/files#r704811693) may apply here too